layers/core_checks/cc_shader.cpp

```diff
@@ -1875,21 +1875,48 @@ bool CoreChecks::ValidateTransformFeedbackDecorations(const SHADER_MODULE_STATE
+bool CoreChecks::ValidateWorkgroupSharedMemory(const SHADER_MODULE_STATE &module_state, VkShaderStageFlagBits stage,
+                                               uint32_t total_workgroup_shared_memory) const {
     bool skip = false;
 
     // If not found before with spec constants, find here
-    if (total_shared_size == 0) {
-        total_shared_size = module_state.CalculateComputeSharedMemory();
+    if (total_workgroup_shared_memory == 0) {
+        total_workgroup_shared_memory = module_state.CalculateWorkgroupSharedMemory();
     }
 
-    if (total_shared_size > phys_dev_props.limits.maxComputeSharedMemorySize) {
-        skip |=
-            LogError(module_state.vk_shader_module(), "VUID-RuntimeSpirv-Workgroup-06530",
-                     "Shader uses %" PRIu32
-                     " bytes of shared memory, more than allowed by physicalDeviceLimits::maxComputeSharedMemorySize (%" PRIu32 ")",
-                     total_shared_size, phys_dev_props.limits.maxComputeSharedMemorySize);
+    switch (stage) {
+        case VK_SHADER_STAGE_COMPUTE_BIT: {
+            if (total_workgroup_shared_memory > phys_dev_props.limits.maxComputeSharedMemorySize) {
+                skip |= LogError(
+                    module_state.vk_shader_module(), "VUID-RuntimeSpirv-Workgroup-06530",
+                    "Shader uses %" PRIu32
+                    " bytes of shared memory, more than allowed by physicalDeviceLimits::maxComputeSharedMemorySize (%" PRIu32 ")",
+                    total_workgroup_shared_memory, phys_dev_props.limits.maxComputeSharedMemorySize);
+            }
+            break;
+        }
+        case VK_SHADER_STAGE_MESH_BIT_EXT: {
+            if (total_workgroup_shared_memory > phys_dev_ext_props.mesh_shader_props_ext.maxMeshSharedMemorySize) {
+                skip |= LogError(module_state.vk_shader_module(), "VUID-RuntimeSpirv-maxMeshSharedMemorySize-08754",
+                                 "Shader uses %" PRIu32
+                                 " bytes of shared memory, more than allowed by maxMeshSharedMemorySize (%" PRIu32 ")",
+                                 total_workgroup_shared_memory, phys_dev_ext_props.mesh_shader_props_ext.maxMeshSharedMemorySize);
+            }
+            break;
+        }
+        case VK_SHADER_STAGE_TASK_BIT_EXT: {
+            if (total_workgroup_shared_memory > phys_dev_ext_props.mesh_shader_props_ext.maxTaskSharedMemorySize) {
+                skip |= LogError(module_state.vk_shader_module(), "VUID-RuntimeSpirv-maxTaskSharedMemorySize-08759",
+                                 "Shader uses %" PRIu32
+                                 " bytes of shared memory, more than allowed by maxTaskSharedMemorySize (%" PRIu32 ")",
+                                 total_workgroup_shared_memory, phys_dev_ext_props.mesh_shader_props_ext.maxTaskSharedMemorySize);
+            }
+            break;
+        }
+        default:
+            break;
```

Comment:
`assert` on `default`.

```suggestion
        default:
            assert(false);
            break;
```